### ISSUE TITLE
Keras Checkpoint Callback's `mode` argument needs clearer documentation

### ISSUE THREAD

**Issue by peterkim95** [tensorflow/tensorflow#40604]
In the documentation for the ModelCheckpoint class, the argument `mode` is described as such:
> ... In `auto` mode, the direction is automatically inferred from the name of the monitored quantity.

Such a description leads to an overestimation of this module's inference capabilities! In my case, I was training a model with `monitor='val_auc'` and `mode='auto'`, believing that TF was setting smartly setting mode as `'max'` as one typically wants to maximize the AUC just like accuracy.
However, I quickly realized that the model was only checkpointing when the AUC was decreasing. I looked through the source code to find out why:
https://github.com/tensorflow/tensorflow/blob/c159f1599548428660c80dada924d69f269384a3/tensorflow/python/keras/callbacks.py#L1195
The algorithm sets mode as `'max'` only when the given metric string contained `'acc'` or when it started with `'fmeasure'` - everything else is set to `'min'`, including for `'val_auc'`.

In my opinion, this inference logic is extremely arbitrary. The string check of `'acc'`or `'fmeasure'` should be either directly mentioned in the documentation, or replaced all-together with a more capable algorithm. Another suggestion is to have a default optimization direction for each metric class and the ModelCheckpoint class can simply refer to that when `mode` is not set. Additionally, one could get rid of `'auto'` as an option since the optimization direction for every metric is clear and obvious from the practitioner's point of view.

**Comment by shahules786** [comment]
I have also encountered the same issue @peterkim95 .

**Comment by shahules786** [comment]
I like to work on this, Either I can change the documentation or modify the code not to support other metrics than 'acc' for `method =auto`

**Comment by ymodak** [comment]
The docs are now updated with commit [e82e381](https://github.com/tensorflow/tensorflow/commit/e82e381c5452e68918bef612ee324a4415108802). Thanks!
